RecoParticleFlow/PFClusterTools/src/PFResolutionMap.cc

```diff
@@ -306,30 +225,35 @@ double PFResolutionMap::minimum(double a,double b){
 //compute the unsigned distance to the closest phi-crack in the barrel
 double PFResolutionMap::dCrackPhi(double phi, double eta){
 
-  static double pi= M_PI;// 3.14159265358979323846;
+  constexpr double pi= M_PI;// 3.14159265358979323846;
+  constexpr double twopi= 2.*pi;
+  constexpr double twopiO18= pi/9;
   
   //Location of the 18 phi-cracks
-  static std::vector<double> cPhi;
-  cPhi.resize(18,0);
-  cPhi[0]=2.97025;
-  for(unsigned i=1;i<=17;i++) cPhi[i]=cPhi[0]-2*i*pi/18;
+  constexpr double c0 = 2.97025;
+  constexpr std::array<double,18> cPhi {{c0, 
+					c0-twopiO18, c0-2*twopiO18, c0-3*twopiO18, c0-4*twopiO18,
+					c0-5*twopiO18, c0-6*twopiO18, c0-7*twopiO18, c0-8*twopiO18,
+					c0-9*twopiO18, c0-10*twopiO18, c0-11*twopiO18, c0-12*twopiO18,
+					c0-13*twopiO18, c0-14*twopiO18, c0-15*twopiO18, c0-16*twopiO18,
+					c0-17*twopiO18}};
```

Comment:
this compile as well
```
constexpr
std::array<float,18> cPhiGener() {
  std::array<float,18> cPhi{0};
  cPhi[0]=2.97025;
  for(unsigned i=1;i<=17;i++) cPhi[i]=cPhi[0]-2*i*M_PI/18;
  return cPhi;
}

constexpr
std::array<float,18> cPhi = cPhiGener();
```

Reply:
nice. I will change.
Thanks.

Reply:
so, this didn't work.
I guess it's c++17 and implies gcc 7.0.

Reply:
indeed. I tested with my version of gcc (8.1)